docs/reference/stores-guide/accumulo.md

```diff
@@ -0,0 +1,320 @@
+```bash
+> offline -t table1
+> clone table table1 table1-backup
+> offline -t table1-backup
+
+# Do your upgrades
+#   - deploy new gaffer jars to Accumulo's class path on each node in your cluster
+#   - run the AddUpdateTableIterator class to update table1
+
+> online -t table1
+
+# Check table1 is still healthy:
+#   - run a query and check the iterators are successfully aggregating and filtering elements correctly.
+
+> droptable -t table1-backup
+```
+
+You will need to run the AddUpdateTableIterator utility providing it with your graphId, schema and store properties.
+If you run it without any arguments it will tell you how to use it.
+
+```bash
+java -cp [path to your jar-with-dependencies].jar uk.gov.gchq.gaffer.accumulostore.utils.AddUpdateTableIterator
+```
+
+## Troubleshooting
+
+**Data hasn't appeared after I performed a bulk import**
+
+Accumulo's UI often shows that there are zero entries in a table after a bulk import. This is generally because Accumulo does not know how many entries have been added until it has performed a major compaction. Open the Accumulo shell, change to the table you specified in your Accumulo properties file, and type `compact`. You should see compactions starting for that table in Accumulo's UI, and the number of entries should then start to increase.
+
+If this has not solved the problem, look at the logs of your bulk import MapReduce job and check that the number of entries output by both the Mapper and the Reducer was positive.
+
+Next check that the elements you generate pass your validation checks.
+
+If all the above fails, try inserting a small amount of data using `AddElements` to see whether the problem is your bulk import job or your data generation.
+
+**Queries result in no data**
+
+Check that you have the correct authorisations to see the data you inserted. Check with the administrator of your Accumulo cluster.
+
+**Spark operations are slow**
+
+Try using a batch scanner to read the data from the tablet server. To enable this for the `GetRDDOfAllElements` or `GetJavaRDDOfAllElements` operation, set the `gaffer.accumulo.spark.rdd.use_batch_scanner` option to true. `GetRDDOfElements` and `GetJavaRDDOfElements` use a batch scanner by default.
+
+If you still don't see a significant improvement, try increasing the value of the `table.scan.max.memory` setting in Accumulo for your table.
+
+**Running accumulo-store Integration Tests and getting error: Error BAD_AUTHORIZATIONS for user root on table integrationTestGraph(ID:1l)**
+
+This means you have correctly set your user (in this case user 'root') in store.properties as ```accumulo.user=root``` however you have not set the correct scan authorisations for the user 'root' required by the integration tests.
+
+If you have the accumulo cluster shell running, you can set these scan auths directly from the shell by entering the following command:
+
+```
+root@instance> setauths -u root -s vis1,vis2,publicVisibility,privateVisibility,public,private
```

Comment:
Can probably take out the `root@instance>` part of this line?

Reply:
I think it's there to indicate that you're at the Accumulo prompt and a root user. I'm not a big fan of it being included, especially when there's a copy to clipboard button. I'll change this.